include/picongpu/plugins/Emittance.hpp

```diff
@@ -0,0 +1,670 @@
+                        std::find(
+                            allowedFilters.begin(),
+                            allowedFilters.end(),
+                            filterName
+                        ) == allowedFilters.end()
+                    )
+                    {
+                        throw std::runtime_error( name + ": unknown filter '" + filterName + "'" );
+                    }
+                }
+            }
+
+            size_t getNumPlugins() const
+            {
+                return notifyPeriod.size();
+            }
+
+            std::string getDescription() const
+            {
+                return description;
+            }
+
+            std::string getOptionPrefix() const
+            {
+                return prefix;
+            }
+
+            std::string getName() const
+            {
+                return name;
+            }
+
+            std::string const name = "CalcEmittance";
+            //! short description of the plugin
+            std::string const description = "calculate the emittance of a species";
+            //! prefix used for command line arguments
+            std::string const prefix = ParticlesType::FrameType::getName( ) + std::string( "_emittance" );
+        };
+
+        //! must be implemented by the user
+        static std::shared_ptr< plugins::multi::IHelp > getHelp()
+        {
+            return std::shared_ptr< plugins::multi::IHelp >( new Help{ } );
+        }
+
+        CalcEmittance(
+            std::shared_ptr< plugins::multi::IHelp > & help,
+            size_t const id,
+            MappingDesc* cellDescription
+        ) :
+            m_help( std::static_pointer_cast< Help >(help) ),
+            m_id( id ),
+            m_cellDescription( cellDescription )
+        {
+            filename = m_help->getOptionPrefix() + "_" + m_help->filter.get( m_id ) + ".dat";
+
+            // decide which MPI-rank writes output
+            writeToFile = reduce.hasResult( mpi::reduceMethods::Reduce( ) );
+
+            // create 4 ints on gpu and host
+            gSum = new GridBuffer<
+                float_64,
+                DIM1
+            >( DataSpace< DIM1 >( 4 ) );
+
+            // only MPI rank that writes to file
+            if( writeToFile )
+            {
+                // open output file
+                outFile.open(
+                    filename.c_str( ),
+                    std::ofstream::out | std::ostream::trunc
+                );
+
+                // error handling
+                if( !outFile )
+                {
+                    std::cerr <<
+                        "Can't open file [" <<
+                        filename <<
+                        "] for output, diasble plugin output. " <<
+                        std::endl;
+                    writeToFile = false;
+                }
+
+                // create header of the file
+                outFile << "#step sum of x² sum of ux² sum of (x*ux)² number of electrons" << " \n";
```

Comment:
please use underscore `_` instead of spaces for grouped element description, at the end you must have 5 described  elements in the header. Please use only ASCII symbols

e.g. `#step sum_of_x_square sum_of_u_times_x_square ...`